src/builder/edit_message.rs

```diff
@@ -254,6 +254,19 @@ impl Builder for EditMessage {
     ) -> Result<Self::Built> {
         self.check_length()?;
 
+        #[cfg(feature = "cache")]
+        if let Some(user_id) = ctx.2 {
+            {
+                if let Some(cache) = cache_http.cache() {
+                    let reference_builder = EditMessage::new().suppress_embeds(true);
+
+                    if user_id != cache.current_user().id && self != reference_builder {
+                        return Err(Error::Model(ModelError::InvalidUser));
+                    }
+                }
+            }
```

Comment:
```suggestion
                if let Some(cache) = cache_http.cache() {
                    let reference_builder = EditMessage::new().suppress_embeds(true);

                    if user_id != cache.current_user().id && self != reference_builder {
                        return Err(Error::Model(ModelError::InvalidUser));
                    }
                }
```

You can delete the inner scope.

Reply:
Thank you for the suggestion